### PR TITLE
Enable new `upload-documents` task

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Create document upload job](https://tools.hmcts.net/jira/browse/BPS-717)

### Change description ###

Enabling new task introduced in #1191. Will link this one in flux PR in a bit

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: charts/bulk-scan-processor/values.yaml [hunk 1]
```diff
@@ -15,6 +15,8 @@ java:
     OCR_VALIDATION_URL_FINREM: http://finrem-cos-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
     SCAN_ENABLED: "true"
     SCAN_DELAY: "4000"
+    UPLOAD_TASK_ENABLED: "true"
+    UPLOAD_TASK_DELAY: "4000"
```
Comment (nitinprabhu) [inline charts/bulk-scan-processor/values.yaml:19]: may be on PR you want less delay ?

Comment (doncem) [reply charts/bulk-scan-processor/values.yaml:19]: true, haven't checked preview template ;)

Comment (doncem) [reply charts/bulk-scan-processor/values.yaml:19]: hm. there's nothing there. deletion only amended to run every 10 seconds. maybe i'll just add there a delay of 2s

Comment (lgonczar) [reply charts/bulk-scan-processor/values.yaml:19]: https://github.com/hmcts/cnp-flux-config/pull/3139 already in place, nice.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/UploadEnvelopeDocumentsTask.java [hunk 2]
```diff
@@ -35,6 +42,8 @@ public UploadEnvelopeDocumentsTask(
         this.uploadService = uploadService;
     }
 
+    @Scheduled(fixedDelayString = "${scheduling.task." + TASK_NAME + ".delay}")
+    @SchedulerLock(name = TASK_NAME) // so to not upload documents multiple times
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/UploadEnvelopeDocumentsTask.java:46]: Should the task name be the name of the reupload task until it's gone?

**Comment by lgonczar** [comment]
Looks good. I have just one question about scheduler lock.

In the future, it would be good to acquire locks on blobs, so that we could have multiple instances uploading envelopes at the same time. This would be an improvement compared to current reupload task.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/UploadEnvelopeDocumentsTask.java (hunk 2 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/UploadEnvelopeDocumentsTask.java:46]: I just realised these two jobs look for envelopes in different statuses. All good.

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/UploadEnvelopeDocumentsTask.java:46]: yep. about to submit a pr for reupload complete re-write ;)

**Comment by doncem** [comment]
locks are acquired in service. but yes - perhaps best to leave for future review once everything settles. it should work as is now with shedlock it's safer for 1st version

**Pull request merged**